packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -681,13 +682,22 @@ function collectSlots(vm: VM, children: VNodes, cmpSlotsMapping: { [key: string]
             continue;
         }
 
-        let slotName = '';
+        let slotName: string | number | boolean = '';
         if (isVBaseElement(vnode)) {
-            slotName = (vnode.data.attrs?.slot as string) ?? '';
+            slotName = vnode.data.attrs?.slot ?? '';
         } else if (isVScopedSlotFragment(vnode)) {
             slotName = vnode.slotName;
         }
 
+        if (!isString(slotName)) {
+            if (process.env.NODE_ENV !== 'production') {
+                logWarn(
+                    `Non-string attribute ${slotName} passed to slot attribute was replaced with an empty string (default slot value). The attribute \`slot\` accepts only string values.`
+                );
+            }
+            slotName = '';
+        }
+
         const vnodes: VNodes = (cmpSlotsMapping[slotName] = cmpSlotsMapping[slotName] || []);
```

Comment:
Let's use a `Map` here (avoid `constructor`/`__proto__`/etc)

Reply:
We already [use `Object.create(null)`](https://github.com/salesforce/lwc/pull/3206/files#diff-8622743d79e146365dd8c59fb85057a09b0f9577d99f065628abdd95d0d7a754R747) while creating an object

Reply:
I prefer `Map` to be explicit, but this is just a taste thing. No biggie.